install/update_94_95.php

```diff
@@ -486,6 +485,68 @@ function update94to95() {
    );
    /** /Clean oprhans documents crontask */
 
+   /** Make software linkable to other itemtypes besides Computers */
+   $migration->renameTable('glpi_computers_softwareversions', 'glpi_items_softwareversions');
+   $migration->addField(
+      'glpi_items_softwareversions',
+      'itemtype',
+      "varchar(100) COLLATE utf8_unicode_ci NOT NULL",
+      [
+         'after'  => 'id',
+         'update' => "'Computer'", // Defines value for all existing elements
+      ]
+   );
+   $migration->changeField(
+      'glpi_items_softwareversions',
+      'computers_id',
+      'items_id',
+      "int(11) NOT NULL DEFAULT '0'"
+   );
+   $migration->changeField('glpi_items_softwareversions', 'is_deleted_computer', 'is_deleted_item', 'bool');
+   $migration->changeField('glpi_items_softwareversions', 'is_template_computer', 'is_template_item', 'bool');
+   $migration->addKey('glpi_items_softwareversions', 'itemtype');
+   $migration->dropKey('glpi_items_softwareversions', 'computers_id');
+   $migration->addKey('glpi_items_softwareversions', 'items_id', 'items_id');
+   $migration->addKey('glpi_items_softwareversions', [
+      'itemtype',
+      'items_id'
+   ], 'item');
+   $migration->dropKey('glpi_items_softwareversions', 'unicity');
+   $migration->addKey('glpi_items_softwareversions', [
+      'itemtype',
+      'items_id',
+      'softwareversions_id'
+   ], 'unicity', 'UNIQUE');
+
+   $migration->renameTable('glpi_computers_softwarelicenses', 'glpi_items_softwarelicenses');
+   $migration->addField(
+      'glpi_items_softwarelicenses',
+      'itemtype',
+      "varchar(100) COLLATE utf8_unicode_ci NOT NULL",
+      [
+         'after'  => 'id',
+         'update' => "'Computer'", // Defines value for all existing elements
+      ]
+   );
+   $migration->changeField(
+      'glpi_items_softwarelicenses',
+      'computers_id',
+      'items_id',
+      "int(11) NOT NULL DEFAULT '0'"
+   );
+   $migration->addKey('glpi_items_softwarelicenses', 'itemtype');
+   $migration->dropKey('glpi_items_softwarelicenses', 'computers_id');
+   $migration->addKey('glpi_items_softwarelicenses', 'items_id', 'items_id');
+   $migration->addKey('glpi_items_softwarelicenses', [
+      'itemtype',
+      'items_id'
+   ], 'item');
+
+   $migration->changeField('glpi_configs', 'purge_computer_software_install',
```

Comment:
This is not a field, but a value.
Should be something like that:
```php
   $migration->addPostQuery(
      $DB->buildUpdate(
         'glpi_configs',
         ['name' => 'purge_item_software_install'],
         ['name' => 'purge_computer_software_install', 'context' => 'core']
      )
   );
```

Reply:
There is a method in the Migration class to add/change configuration values (Migration::addConfig)

Reply:
I don't see a method for changing config names in Migration (or Config). The addConfig method only accepts name => value pairs and ignores existing configs.

Reply:
I was indeed wrong, existing values are ignored there. addPostQuuery is the best option then ;)